#!/usr/bin/env python
PACKAGE = "elevation_mapping"

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t, int_t

gen = ParameterGenerator()
gen.add("length_in_x",int_t, 0, "map length in x", 10,0,20)
gen.add("length_in_y",int_t, 0, "map length in y", 10,0,20)
gen.add("position_x",int_t, 0, "robot pose in x", 0,-10,10)
gen.add("position_y",int_t, 0, "robot pose in y", 0,-10,10)
gen.add("resolution",double_t, 0, "robot pose in y", 0.01,0.01,0.4)

gen.add("min_variance",double_t, 0, "min_variance",0.0001,0.0,0.004)
gen.add("max_variance",double_t, 0, "max_variance", 10000.0,1.0,100000.0)
gen.add("mahalanobis_distance_threshold",double_t, 0, "mahalanobis_distance_threshold", 0.1,0.0,4.0)
gen.add("multi_height_noise",double_t, 0, "multi_height_noise", 0.00002,0.0,0.4)


exit(gen.generate("elevation_mapping", "elevation_mapping", "ElevationMapPlugin"))


